planning/behavior_path_planner/src/turn_signal_decider.cpp

```diff
@@ -148,10 +181,11 @@ boost::optional<TurnSignalInfo> TurnSignalDecider::getIntersectionTurnSignalInfo
 
     const auto & turn_signal_info = signal_queue.front();
     const auto & required_end_point = turn_signal_info.required_end_point;
-    const size_t nearest_seg_idx =
-      motion_utils::findNearestSegmentIndex(path.points, required_end_point);
+    const size_t nearest_seg_idx = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+      path.points, required_end_point, nearest_dist_threshold, nearest_yaw_threshold);
```

Comment:
Is it okay to use find"First"NearestSegmentIndexWithSoftConstraints?
This function is designed to find "ego"'s nearest segment index.